solr/core/src/test/org/apache/solr/cloud/api/collections/CollectionReloadTest.java

```diff
@@ -86,4 +96,107 @@ public void testReloadedLeaderStateAfterZkSessionLoss() throws Exception {
 
     log.info("testReloadedLeaderStateAfterZkSessionLoss succeeded ... shutting down now!");
   }
+
+  @Repeat(iterations = 5)
+  public void testCreateReloadDeleteAllNrt() throws Exception {
+    testCreateReloadDelete("testCreateReloadDeleteAllNrt", 3, 0, 0);
+  }
+
+  @Repeat(iterations = 5)
+  public void testCreateReloadDeleteAllTlog() throws Exception {
+    testCreateReloadDelete("testCreateReloadDeleteAllTlog", 0, 3, 0);
+  }
+
+  @Repeat(iterations = 5)
+  public void testCreateReloadDeletePull() throws Exception {
+    testCreateReloadDelete("testCreateReloadDeletePull", 0, 1, 2);
+  }
+
+  private void testCreateReloadDelete(
+      String collectionName, int nrtReplicas, int tlogReplicas, int pullReplicas) throws Exception {
+    int numShards = 3;
+    createCollection(numShards, nrtReplicas, tlogReplicas, pullReplicas, collectionName);
+    boolean reloaded = false;
+    while (true) {
+      waitForState(
+          "Timeout waiting for collection " + collectionName,
+          collectionName,
+          clusterShape(numShards, numShards * (nrtReplicas + tlogReplicas + pullReplicas)));
+      if (reloaded) {
+        break;
+      } else {
+        // reload
+        assertSuccessfulAdminRequest(
+            CollectionAdminRequest.reloadCollection(collectionName)
+                .process(cluster.getSolrClient()));
+        reloaded = true;
+      }
+    }
+    assertSuccessfulAdminRequest(
+        CollectionAdminRequest.deleteCollection(collectionName).process(cluster.getSolrClient()));
+  }
+
+  private void assertSuccessfulAdminRequest(CollectionAdminResponse response) {
+    assertEquals("Unexpected response status", 0, response.getStatus());
+    assertTrue("Unsuccessful response: " + response, response.isSuccess());
+  }
+
+  private void createCollection(
+      int numShards, int nrtReplicas, int tlogReplicas, int pullReplicas, String collectionName)
+      throws SolrServerException, IOException {
+    switch (random().nextInt(3)) {
```

Comment:
Seems weird these create strings - don't we have admin API wrappers around these? We shouldn't need to build straight httpget/httppost commands...